client/src/containers/blog/Blog.js

```diff
@@ -1,27 +1,94 @@
-import React from 'react';
-import { Route, Switch, useRouteMatch } from 'react-router-dom';
-import LinkButton from '../../components/LinkButton';
-import BlogPost from '../blog-post/BlogPost';
+import React, { useEffect, useState } from 'react';
+import { Col, Row, Spin, Typography } from 'antd';
+import { Trans } from '@lingui/macro';
+import BlogItem from '../../components/BlogItem/BlogItem';
+import config from '../../config';
+import LoadMore from '../../components/LoadMoreArticlesLink/LoadMoreArticlesLink';
+
+const { Title, Paragraph } = Typography;
+
+const { POSTS_URL } = config;
+
+const LIMIT = 3;
 
 const Blog = () => {
-  const match = useRouteMatch();
+  const [state, setState] = useState({
+    posts: [],
+    requestError: false,
+    loading: true,
+    index: 0,
+    showMore: true,
+  });
+
+  const loadPosts = async () => {
+    try {
+      const res = await fetch(`${POSTS_URL}`);
+      const posts = await res.json();
+      const newIndex = state.index + LIMIT;
+      const newList = [...state.posts, ...posts.slice(state.index, newIndex)];
+      const newShowMore = newIndex < posts.length;
+      setState({
+        ...state,
+        posts: newList,
+        loading: false,
+        index: newIndex,
+        showMore: newShowMore,
+        requestError: false,
+      });
+    } catch (err) {
+      setState({
+        ...state,
+        posts: [],
+        requestError: true,
+        loading: false,
+        index: 0,
+        showMore: true,
+      });
+    }
+  };
+
+  useEffect(() => {
+    loadPosts();
+  }, []);
```

Comment:
`loadPosts` needs to be specified in the dependencies array (or declared directly in the effect).
You might need to wrap `loadPosts` in useCallback.

Reply:
I am having trouble understanding why this is necessary because now it works fine:
- `loadPosts` is called once when component is mounted.
- `loadPosts` is called every time the `LoadMore` component is clicked.

Now if I add `loadPosts` as a dependency:
- I need to use `useCallback` (as you mentioned) on `loadPosts`, without it the function being different on each render and `useEffect` getting called again and again. 
- If I use an empty array as dependency for `useCallback`, `LoadMore` component will not work anymore because `loadPosts` was not recreated and the state is the one from the initial render.
- If I use the state as dependency, the function is recreated each time the state changes but that means `useEffect` is called as well resulting in infinite re-rendering.

Another option is empty array for `useEffect`, wrap `loadPosts` with `useCallback` and provide the state as a dependency this way `loadPosts` will be recreated only if the state changes (if this is indeed the reason behind your request).